dcrtimed/backend/filesystem/filesystem.go

```diff
@@ -588,6 +588,71 @@ func (fs *FileSystem) getDigest(ts int64, current *leveldb.DB, digest [sha256.Si
 		}
 	}
 
+	// Lookup in previous not flushed dirs
+	// Get Dirs.
+	files, err := ioutil.ReadDir(fs.root)
+	if err != nil {
+		return gdme, err
+	}
+	// Collect relevant dirs.
+	nowDir := now.Format(fStr)
+	dirs := make([]string, 0, len(files))
+	for _, file := range files {
+		// Skip global db.
+		if file.Name() == globalDBDir {
+			continue
+		}
+		if !file.IsDir() {
+			continue
+		}
+		// Skip current timestamp.
+		if file.Name() == nowDir {
+			continue
+		}
+
+		dirs = append(dirs, file.Name())
+	}
+
+	// Reverse sort work.
+	sort.Sort(sort.Reverse(sort.StringSlice(dirs)))
+
+	// Walk directories backwards until we find a flushed database. At
+	// this point we know we are caught up.
+	foundP := false
+	for _, dir := range dirs {
+		timestamp, err := time.Parse(fStr, dir)
+		if err != nil {
+			continue
+		}
+		dirTs := timestamp.Unix()
+		if fs.isFlushed(dirTs) {
+			// We hit a flushed dir so we should be done.
+			break
+		}
+
+		// Open dir database
+		dirDb, err := fs.openRead(dirTs)
+		if err != nil {
+			return gdme, err
+		}
+		defer dirDb.Close()
+		foundP, err = dirDb.Has(digest[:], nil)
+		if err != nil {
+			return gdme, err
+		}
+		if foundP {
+			gdme.ErrorCode = backend.ErrorOK
+			gdme.AnchoredTimestamp = 0 // Dir not anchored yet
+
+			// Override error code during testing
+			if fs.testing {
+				gdme.ErrorCode = foundPrevious
```

Comment:
Why are you swallowing errors when testing? Doesn't that defeat the whole point of testing?

Reply:
This pattern is used in other places as well.  The intention is to be able to differentiate between _found_ ErrorCodes, as we have three different cases:

1. Found in current cycle dir
2. Found in previous not anchored yet cycle.
3. Anchored and found in global db.

Reply:
From API perspective all the above cases will return one errorcode.